VRDR.Messaging/CodingResponseMessage.cs

```diff
@@ -140,14 +140,267 @@ public enum RaceCode
+                    }
+                    Record.Add("rec_dy", new UnsignedInt((int)value));
+                }
+            }
+        }
+
+        /// <summary>NCHS Receipt Date Year (Numeric, Greater than year of death or blank)</summary>
+        public uint? NCHSReceiptYear
+        {
+            get
+            {
+                return (uint?)Record.GetSingleValue<UnsignedInt>("rec_yr")?.Value;
+            }
+            set
+            {
+                Record.Remove("rec_yr");
+                if (value != null)
+                {
+                    if (DeathYear != null && value < DeathYear) {
+                        throw new ArgumentException("NCHS Receipt Year must be greater than Death Year, or null");
+                    }
+                    Record.Add("rec_yr", new UnsignedInt((int)value));
+                }
+            }
+        }
+
+        /// <summary>Manner of Death Enum</summary>
+        public enum MannerOfDeathEnum
+        {
+            /// <summary>Natural cause of death</summary>
+            Natural,
+            /// <summary>Accident cause of death</summary>
+            Accident,
+            /// <summary>Suicide cause of death</summary>
+            Suicide,
+            /// <summary>Cause of death is pending investigation</summary>
+            PendingInvestigation,
+            /// <summary>Cause of death could not be determined</summary>
+            CouldNotBeDetermined
+        }
+
+        /// <summary>Manner of Death (or null)</summary>
+        public MannerOfDeathEnum? MannerOfDeath
+        {
+            get
+            {
+                string value = Record.GetSingleValue<FhirString>("manner")?.Value;
+                if(value == null)
+                {
+                    return null;
+                }
+                else
+                {
+                    Enum.TryParse<MannerOfDeathEnum>(value, out MannerOfDeathEnum code);
+                    return code;
+                }
+            }
+            set
+            {
+                Record.Remove("manner");
+                if(value != null)
+                {
+                    Record.Add("manner", new FhirString(value.ToString()));
+                }
+            }
+        }
+
+        /// <summary>Intentional reject (1-5, 9 or null)</summary>
+        public uint? IntentionalReject
+        {
+            get
+            {
+                return (uint?)Record.GetSingleValue<UnsignedInt>("int_rej")?.Value;
+            }
+            set
+            {
+                Record.Remove("int_rej");
+                if (value != null)
+                {
+                    if (value != 9 || value > 5 || value < 1) {
+                        throw new ArgumentException("Intentional Reject must equal 1-5, 9, or null.");
+                    }
+                    Record.Add("int_rej", new UnsignedInt((int)value));
+                }
+            }
+        }
```

Comment:
Question: should some of these be of the form

```json
{
    "name": "int_rej",
    "valueCoding": {
        "system": "<some code system>",
        "code": "<1-5, 9>"
    }
}
```

instead of

```json
{
    "name": "int_rej",
    "valueUnsignedInt": "<1-5, 9>"
}
```

Reply:
I think so, see previous comments.

Reply:
I've gone ahead and changed this to be a Code, one question I have is should I still have logic to check if it is a valid value or should I rely on the user to properly implement the code system?

Reply:
The VRDR library leaves it up to the user to ensure codes are valid. I'd recommend doing the same for VRDR.Messaging.